File: src/platform/update.ts
import { call, on } from './invoke';
import type { UpdateCheckResult } from './types';

export const updateApi = {
  status: () => call<UpdateCheckResult>('update_status'),
  check: () => call<UpdateCheckResult>('update_check'),
  install: () => call('update_install'),
  onDownloaded: (cb: (info: { version: string }) => void) => on<{ version: string }>('update:downloaded', cb),
};

File: src-tauri/src/ipc/update.rs
use tauri::AppHandle;

use crate::error::AppResult;
use crate::services::update::{self, UpdateStatus};

#[tauri::command]
pub async fn update_status() -> AppResult<UpdateStatus> {
    Ok(update::status())
}

#[tauri::command]
pub async fn update_check(app: AppHandle) -> AppResult<UpdateStatus> {
    Ok(update::check(app, true).await)
}

#[tauri::command]
pub async fn update_install() -> AppResult<()> {
    update::install()
}
